src/bootstrap/src/core/builder.rs

```diff
@@ -1671,7 +1701,7 @@ impl<'a> Builder<'a> {
             .env("RUSTDOC", self.bootstrap_out.join("rustdoc"))
             .env(
                 "RUSTDOC_REAL",
-                if cmd == "doc" || cmd == "rustdoc" || (cmd == "test" && want_rustdoc) {
+                if cmd == "doc" || cmd == "rustdoc" || (cmd.ends_with("test") && want_rustdoc) {
```

Comment:
It's probably worth adding a comment explaining why we use `ends_with` here. For those unfamiliar with the Miri context (which is quite normal as we manage too many tool builds in addition to compiler and library), it could be difficult to understand the rationale behind this.

Reply:
Makes sense.

The alternative would be to not use `miri-$cmd` but add a separate `miri: bool` flag (or a 2-value enum to avoid bare bools). Not sure which way you prefer.

Reply:
I believe the best approach would be to create an enum for `cmd` and handle all these string hacks within the implementation for that enum which can fairly simplify this overly generic function. If you prefer not to handle that in this PR, you can leave a FIXME note on top of `cmd` arg for this function.

Reply:
Note that with Miri, it's not just an enum -- it's an enum plus a boolean. I guess we could have `MiriTest` and `MiriSetup` variants but then there's a risk some places will treat Test different from MiriTest.

I will add a FIXME.